morpheus/models/dfencoder/autoencoder.py

```diff
@@ -737,251 +741,132 @@ def _create_stat_dict(self, a):
-            If rank and world_size not provided in distributed training mode.
-        TypeError
-            If train_data is not a pandas dataframe or a torch.utils.data.DataLoader or a torch.utils.data.Dataset in distributed training mode.
-        """
-        if not self.distributed_training:
-            if not isinstance(train_data, pd.DataFrame):
-                raise TypeError("`train_data` needs to be a pandas dataframe in centralized training mode."
-                                f" `train_data` is currently of type: {type(train_data)}")
-
-            self._fit_centralized(
-                df=train_data,
-                epochs=epochs,
-                val=val_data,
-                run_validation=run_validation,
-                use_val_for_loss_stats=use_val_for_loss_stats,
-            )
+        if (isinstance(data, pd.DataFrame)):
+            dset = DataframeDataset(df=data)
+        elif (isinstance(data, DFEncoderDataLoader)):
+            dset = data.dataset
         else:
-            # distributed training requires rank and world_size
-            if rank is None or world_size is None:
-                raise ValueError('`rank` and `world_size` must be provided for distributed training.')
-
-            if not isinstance(train_data, (pd.DataFrame, torch.utils.data.DataLoader, torch.utils.data.Dataset)):
-                raise TypeError(
-                    "`train_data` needs to be a pandas DataFrame, a DataLoader, or a Dataset in distributed training mode."
-                    f" `train_data` is currently of type: {type(train_data)}")
-
-            self._fit_distributed(
-                train_data=train_data,
-                epochs=epochs,
-                val_data=val_data,
-                run_validation=run_validation,
-                use_val_for_loss_stats=use_val_for_loss_stats,
-                rank=rank,
-                world_size=world_size,
-            )
+            dset = data
 
-    def _fit_centralized(self, df, epochs=1, val=None, run_validation=False, use_val_for_loss_stats=False):
-        """Does training in a single process on a single GPU.
+        if (train):
+            dset = self._transform_dataset_for_training(dset)
+        else:
+            dset = self._transform_dataset_for_validation(dset)
 
-        Parameters
-        ----------
-        df : pandas.DataFrame
-            Data used for training.
-        epochs : int, optional
-            Number of epochs to run training, by default 1.
-        val : pandas.DataFrame, optional
-            Optional pandas dataframe for validation or loss stats, by default None.
-        run_validation : bool, optional
-            Whether to collect validation loss for each epoch during training, by default False.
-        use_val_for_loss_stats : bool, optional
-            Whether to use the validation set for loss statistics collection (for z score calculation), by default False.
+        return dset if not is_loader else data
```

Comment:
We need to transform the dataset whether its created by us, passed by a user, or passed inside of a DataLoader, because we modify it for training vs validation and want to be sure everything is in a known consistent state prior to training.

I'll add a comment.

Reply:
So true transformations are mutative. That makes sense. It’s just not immediately obvious since we’re also sometimes returning a value.